org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -137,6 +138,40 @@ public void linear_instruction_sequence_should_ignore_instructions_when_filter_i
 		assertLine(1002, 0, 1, 0, 0);
 	}
 
+	// === Scenario: method invocation after zero line number
+
+	private void createZeroLineNumber() {
+		final Label l0 = new Label();
+		method.visitLabel(l0);
+		method.visitLineNumber(1001, l0);
+		method.visitInsn(Opcodes.NOP);
+		final Label l1 = new Label();
+		method.visitLabel(l1);
+		method.visitLineNumber(0, l1);
+		method.visitMethodInsn(Opcodes.INVOKESTATIC, "Foo", "foo", "()V",
+				false);
+		method.visitInsn(Opcodes.NOP);
+		final Label l2 = new Label();
+		method.visitLabel(l2);
+		method.visitLineNumber(1002, l2);
+		method.visitInsn(Opcodes.RETURN);
+	}
+
+	/**
+	 * @see org.jacoco.core.internal.flow.LabelFlowAnalyzer#visitLineNumber(int,
+	 *      Label)
+	 * @see org.jacoco.core.internal.instr.ZeroLineNumberTest
+	 */
+	@Test
+	public void zero_line_number_should_create_1_probe() {
+		createZeroLineNumber();
+		runMethodAnalzer();
+		assertEquals(1, nextProbeId);
+		// workaround for zero line number can be removed if needed
+		// during change of exec file version
+		assertEquals(0x1007, ExecutionDataWriter.FORMAT_VERSION);
```

Comment:
I love this idea! 👍

Reply:
@marchof I have a strong feeling of déjà vu... oh, no actually it indeed happened - https://github.com/jacoco/jacoco/pull/636#discussion_r159135244 😆